### PR TITLE
Implemented storage key-value class

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,47 @@
...
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            for (int i = 0; i < size; i++) {
+                if ((keys[i] == key)
+                        || (keys[i] != null && keys[i].equals(key))) {
+                    values[i] = value;
+                }
+            }
```
Comment (irynamekh) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:24]: here the logic is repeated, the same as in the get() method. Try to move the repeating logic into a separate private method

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,47 @@
...
+        } else {
+            throw new RuntimeException("Storage is already filled. Can not add new elements");
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == key)
+                    || (keys[i] != null && keys[i].equals(key))) {
```
Comment (irynamekh) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:38]: everything is correct, but I prefer to use Objects.equals(), then the code looks clearer. This is just an idea, you can not change anything here.

**Review by irynamekh — CHANGES_REQUESTED** [review]
if my fixes break something, please tell me

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (Skarbey) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:38]: Thanks. I agree but in common misstakes Objects are restricted

**Review by irynamekh — APPROVED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int MAXIMUM_ARRAY_SIZE = 10;
```
Comment (nacenik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:6]:
```suggestion
    public static final int MAXIMUM_STORAGE_SIZE = 10;
```

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it)

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,47 @@
...
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private int getIndexByKey(K key) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == key) || (keys[i] != null && keys[i].equals(key))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:43]:
```suggestion
            if (keys[i] == key || keys[i] != null && keys[i].equals(key)) {
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**